### PR TITLE
#1442: add test for OptCached

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Test cases for `OptCached `.

Closes: #1442

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@Graur have a look please

**Review by Graur — APPROVED** [review]
@volodya-lombrozo Good one!
@yegor256 Please, take a look

**Comment by volodya-lombrozo** [comment]
@yegor256 reminder

File: eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     * Get parsed program from resources.
+     * @return XML representation of program.
+     */
+    private static XML program() {
+        final String path = "org/eolang/maven/optimize/main.xml";
+        try {
+            return new XMLDocument(
+                new UncheckedInput(new ResourceOf(path)).stream()
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java:107]: @volodya-lombrozo it's a bad idea to call `.stream()` and not close it later. Don't do it. Instead, use `XMLDocument(path)`.

File: eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Save XML program with hardcoded name to a temp directory.
+     * @param tmp Temporary test directory.
+     * @param xml XML program.
+     * @return Path to saved program.
+     */
+    private static Path save(final Path tmp, final XML xml) {
+        final Path res = tmp.resolve("main.xmir");
+        try {
+            Files.write(res, xml.toString().getBytes(StandardCharsets.UTF_8));
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java:92]: @volodya-lombrozo why can't you use our object `Home`, which knows how to save into file?

File: eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java]: @yegor256 I don't wanna create and save intermediate file in order to use `XMLDocument(path)`, I wanna load xml right from resources, by that reason I've used the next code:
```java
 private static XML program() {
        return new XMLDocument(
            new UncheckedBytes(
                new BytesOf(
                    new ResourceOf("org/eolang/maven/optimize/main.xml")
                )
            ).asBytes()
        );
    }
```

File: eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java (hunk 2 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java]: @yegor256 done.

File: eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     * @param xml XML program.
+     * @return Path to saved program.
+     */
+    private static Path save(final Path tmp, final XML xml) {
+        final Path path = Paths.get("main.xmir");
+        final Path res = tmp.resolve(path);
+        try {
+            new Home(tmp).save(xml.toString().getBytes(StandardCharsets.UTF_8), path);
+            return res;
+        } catch (final IOException ex) {
```
Comment (yegor256) [inline eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java:97]: @volodya-lombrozo when you _have to_ catch an exception -- it's an indicator of a wrong design. Can't you just let it float up?

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java]: @yegor256 fixed.

**Comment by volodya-lombrozo** [comment]
@yegor256 reminder

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**